[Serve] [Docs] Add Serve REST API Schema to Serve API Docs (#25786)
ray-project/ray: doc/source/serve/package-ref.md

```{eval-rst}
.. autofunction:: ray.serve.batch(max_batch_size=10, batch_wait_timeout_s=0.0)
```

## Serve REST API

### Config

#### Schema

```{eval-rst}
.. autopydantic_model:: ray.serve.schema.ServeApplicationSchema

```

#### REST API

```{eval-rst}
.. http:GET:: "/api/serve/deployments/"
    :noindex:

>         Gets latest config that Serve has received. This config represents the current goal state for the Serve application. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the output schema.

Reviewer: link to "config schema" and don't capitalize. also goes for other instances

Author: Good catch– I made the updates.

```

```{eval-rst}
.. http:PUT:: "/api/serve/deployments/"
    :noindex:

>         Uploads config for Serve application to deploy. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the request schema.

Reviewer: ```suggestion
        Declaratively deploys the Serve application. Starts Serve on the Ray cluster if it's not already running. See Config Schema for the request schema.
```
